File: 3104/6601001123_HW1_2/main.cpp
#include <iostream>

using namespace std ;

class employee {
    public :
        string name ;
        int age ;
        float salary ;
        bool employed ;

        employee() {
            
        }
        employee(string name, int age, float salary, bool employed) {
            this->name = name ;
            this->age = age ;
            this->salary = salary ;
            this->employed = employed ;
        }
};

string getType(string) {return "string";} 
string getType(int) {return "int";}
string getType(float) {return "float";}
string getType(bool) {return "bool";}

int main() {
    employee employees[] = {
        employee("Alice",30,55000.75,true),
        employee("Bob",25,48000.00,false)
    };
    
    cout << boolalpha ;
    for(int i = 0 ; i < sizeof(employees)/sizeof(employees[0]) ; i++) {
        cout << "Employee " << i+1 << endl ;
        cout << "   Name: " << employees[i].name << " ,Type: " << getType(employees[i].name) << endl ;
        cout << "   Age: " << employees[i].age << " ,Type: " << getType(employees[i].name) << endl ;
        cout << "   Salary: " << employees[i].salary << " ,Type: " << getType(employees[i].name) << endl ;
        cout << "   Employed: " << employees[i].employed << " ,Type: " << getType(employees[i].name) << endl ;
    }
    
}

File: 3104/6601001123_HW1_2/main.py
# Each employee is a list with mixed data types: [name, age, salary, employed]
employees = [
    ["Alice", 30, 55000.75, True],
    ["Bob", 25, 48000.00, False]
]

def increaseSalary(employee) :
    employee[2] = employee[2]*110/100
    
# Access and print each employee's data
# for i, emp in enumerate(employees):
#     print(f"Employee {i+1}:")
#     print(f"  Name: {emp[0]} ,Type: {type(emp[0])}")
#     print(f"  Age: {emp[1]},Type: {type(emp[0])}")
#     print(f"  Salary: {emp[2]},Type: {type(emp[0])}")
#     print(f"  Employed: {emp[3]},Type: {type(emp[0])}")
#     print()
print(employees[1][2])
increaseSalary(employees[1])
print(employees[1][2])